components/org.wso2.carbon.identity.sts.passive.ui/src/main/java/org/wso2/carbon/identity/sts/passive/ui/PassiveSTS.java

```diff
@@ -498,6 +501,19 @@ private void process(HttpServletRequest request, HttpServletResponse response,
     private void handleLogoutRequest(HttpServletRequest request, HttpServletResponse response)
             throws IOException, PassiveSTSException {
 
+        // Validate the logout url if the logout wreply validation is enabled.
+        if (Boolean.parseBoolean(IdentityUtil.getProperty(
+                IdentityConstants.STS.PASSIVE_STS_LOGOUT_WREPLY_VALIDATION))) {
+            try {
+                validateLogoutURL(request);
+            } catch (PassiveSTSException e) {
+                log.error(e.getMessage());
+                PassiveSTSUtil.sendToErrorPage(request, response,
+                        ERROR_AUTHENTICATION, ERROR_MSG_LOGOUT_WREPLY_MISMATCH);
+                return;
```

Comment:
Do we need a return?

Reply:
Yes. Otherwise the logout request will proceed even after the error is thrown and cause issues in the redirection to the error page.